### PR TITLE
[2322] Performance improvements to end of cycle jobs

### PR DESCRIPTION
**Pull request by elceebee**
## Context

Having seen the `CancelUnsubmittedApplicationsWorker` perform yesterday, I decided we should make changes to it (for next year) and for the other two jobs that are going to run this end-of-cycle period. 

## Changes proposed in this pull request

- For all three workers (cancel unsubmitted, reject by default, and decline by default), we will batch the applications into smaller jobs so that we don't put as much pressure on a single worker. 
- Added a few more tests

## Guidance to review

Any advice, thoughts for more improvements? 

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [x] API release notes have been updated if necessary
- [x] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [x] Inform data insights team due to database changes
- [x] Make sure all information from the Trello card is in here
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Add PR link to Trello card

### CODE REVIEW

**Comment by elceebee** [comment]
[Tech debt: Improvements to end of cycle jobs](https://trello.com/c/LmMHG3Sy/2322-tech-debt-improvements-to-end-of-cycle-jobs)

File: app/workers/end_of_cycle/cancel_unsubmitted_applications_worker.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        .distinct
+    end
+  end
+
+  class CancelUnsubmittedApplicationsSecondaryWorker
+    include Sidekiq::Worker
+
+    def perform(application_form_ids)
+      application_form_ids.each do |application_form_id|
+        ApplicationForm.find(application_form_id).application_choices.unsubmitted.each do |application_choice|
```
Comment (tomas-stefano) [inline app/workers/end_of_cycle/cancel_unsubmitted_applications_worker.rb:28]: Can we change to not loop through each application_form_ids? That would be N+1 query? There would be 400 queries per job (Application form and application choices query) + 1 update. I wonder why Bullet didn't raise anything. 🤔

**Review by CatalinVoineag — APPROVED** [review]
Looks good

**Review by dcyoung-dev — APPROVED** [review]

**Review by tomas-stefano — APPROVED** [review]
Looks great ✌️

**Pull request merged**